File: app/src/main/java/com/nicrosoft/consumoelectrico/myUtils/ReminderService.java
package com.nicrosoft.consumoelectrico.myUtils;

import androidx.appcompat.app.AppCompatDelegate;

import com.firebase.jobdispatcher.JobParameters;
import com.firebase.jobdispatcher.JobService;
import com.nicrosoft.consumoelectrico.R;
import com.nicrosoft.consumoelectrico.realm.Lectura;
import com.pixplicity.easyprefs.library.Prefs;

import org.joda.time.Days;
import org.joda.time.LocalDate;
import org.joda.time.LocalDateTime;

import java.util.Date;

import io.reactivex.disposables.CompositeDisposable;
import io.realm.Realm;
import io.realm.RealmResults;
import io.realm.Sort;

public class ReminderService extends JobService {
    public static final String JOB_TAG = "ReminderService";
    CompositeDisposable mDisposable = new CompositeDisposable();

    @Override
    public boolean onStartJob(JobParameters job) {
        // Answers the question: "Is there still work going on?"
        //Log.e("EDER", "ReminderService START");
        try {
            Realm realm = Realm.getDefaultInstance();
            RealmResults<Lectura> res = realm.where(Lectura.class)
                    .findAll().sort("fecha_lectura", Sort.DESCENDING);
            if (res.size() > 0) {
                Lectura lectura = res.first();
                if (lectura != null) {
                    int dias_recordatorio = 5;
                    try {
                        dias_recordatorio = Integer.parseInt(Prefs.getString("reminder_after", "7"));
                    } catch (Exception ignored) {
                    }
                    Days days = Days.daysBetween(new LocalDate(lectura.fecha_lectura), new LocalDate(new Date()));
                    if (days.getDays() > dias_recordatorio) {
                        LocalDateTime now = LocalDateTime.now();
                        int hour = now.getHourOfDay();
                        if (hour >= 7 && hour <= 21) {
                            NotificationHelper helper = new NotificationHelper(getApplicationContext());
                            helper.createNotification(getApplicationContext().getString(R.string.reminder), getApplicationContext().getString(R.string.reminder_msg, String.valueOf(days.getDays())));
                        }
                    }
                }
            }
            realm.close();
            return false;
        } catch (Exception f){
            return false;
        }
    }

    @Override
    public boolean onStopJob(JobParameters job) {
        // si el trabajo falla retornar true para reintentar
        return true;
    }
}

File: app/src/main/java/com/nicrosoft/consumoelectrico/utils/ReminderService.kt
package com.nicrosoft.consumoelectrico.utils

import com.firebase.jobdispatcher.JobParameters
import com.firebase.jobdispatcher.JobService
import com.nicrosoft.consumoelectrico.R
import com.nicrosoft.consumoelectrico.realm.Lectura
import com.nicrosoft.consumoelectrico.utils.helpers.NotificationHelper
import com.pixplicity.easyprefs.library.Prefs
import io.realm.Realm
import io.realm.Sort
import org.joda.time.Days
import org.joda.time.LocalDate
import org.joda.time.LocalDateTime
import java.util.*

class ReminderService : JobService() {
    override fun onStartJob(job: JobParameters): Boolean {
        // Answers the question: "Is there still work going on?"
        //Log.e("EDER", "ReminderService START");
        return try {
            val realm = Realm.getDefaultInstance()
            val res = realm.where(Lectura::class.java)
                    .findAll().sort("fecha_lectura", Sort.DESCENDING)
            if (res.size > 0) {
                val lectura = res.first()
                if (lectura != null) {
                    var dias_recordatorio = 5
                    try {
                        dias_recordatorio = Prefs.getString("reminder_after", "7").toInt()
                    } catch (ignored: Exception) {
                    }
                    val days = Days.daysBetween(LocalDate(lectura.fecha_lectura), LocalDate(Date()))
                    if (days.days > dias_recordatorio) {
                        val now = LocalDateTime.now()
                        val hour = now.hourOfDay
                        if (hour in 7..21) {
                            val helper = NotificationHelper(applicationContext)
                            helper.createNotification(applicationContext.getString(R.string.reminder), applicationContext.getString(R.string.reminder_msg, days.days.toString()))
                        }
                    }
                }
            }
            realm.close()
            false
        } catch (f: Exception) {
            false
        }
    }

    override fun onStopJob(job: JobParameters): Boolean {
        // si el trabajo falla retornar true para reintentar
        return true
    }

    companion object {
        const val JOB_TAG = "ReminderService"
    }
}